lib/truffle/io/console.rb

```diff
@@ -45,6 +45,38 @@ def getpass(prompt = nil)
 
 
 class IO
+  def self.console(sym=nil, *args)
+    raise TypeError, "expected Symbol, got #{sym.class}" unless sym.nil? || sym.kind_of?(Symbol)
+
+    # klass = self == IO ? File : self
+    if defined?(@console) # using ivar instead of hidden const as in MRI
+      con = @console
+      # MRI checks IO internals : (!RB_TYPE_P(con, T_FILE) || (!(fptr = RFILE(con)->fptr) || GetReadFD(fptr) == -1))
+      if !con.kind_of?(File) || (con.kind_of?(IO) && (con.closed? || !FileTest.readable?(con)))
+        remove_instance_variable :@console
+        con = nil
+      end
+    end
+
+    if sym
+      if sym == :close
```

Comment:
These 2 `if` can be folded in one.

Reply:
Again, just copied.

Reply:
Yes, but I think we should cleanup code we import like this, since we'll have to maintain it.